x/bank/types/keys.go

```diff
@@ -65,6 +65,12 @@ func AddressAndDenomFromBalancesStore(key []byte) (sdk.AccAddress, string, error
 	return key[1 : addrBound+1], string(key[addrBound+1:]), nil
 }
 
+// CreatePrefixedAccountStoreKey returns the key for the given account and denomination.
+// This method can be used when performing an ABCI query for the balance of an account.
+func CreatePrefixedAccountStoreKey(addr []byte, denom string) []byte {
```

Comment:
could you replace the places we manually create the key with this function for consistency,